src/Lean/Server/AsyncList.lean

```diff
@@ -65,19 +65,23 @@ partial def getAll : AsyncList ε α → List α × Option ε
     | Except.ok tl => tl.getAll
     | Except.error e => ⟨[], some e⟩
 
-/-- Spawns a `Task` waiting on the prefix of elements for which `p` is true. -/
-partial def waitAll (p : α → Bool := fun _ => true) : AsyncList ε α → Task (List α × Option ε)
+/-- Spawns a `Task` returning the prefix of elements up to (including) the first element for which `p` is true. -/
+partial def waitUntil (p : α → Bool) : AsyncList ε α → Task (List α × Option ε)
   | cons hd tl =>
-    if p hd then
-      (tl.waitAll p).map fun ⟨l, e?⟩ => ⟨hd :: l, e?⟩
+    if !p hd then
+      (tl.waitUntil p).map fun ⟨l, e?⟩ => ⟨hd :: l, e?⟩
     else
       .pure ⟨[hd], none⟩
```

Comment:
Ironically enough it already returned this additional element, no change necessary